skglm/solvers/gram.py

```diff
@@ -0,0 +1,96 @@
+import numpy as np
+from numba import njit
+from numpy.linalg import norm
+from celer.homotopy import _grp_converter
+
+from skglm.utils import BST, ST
+
+
+@njit
+def primal(alpha, y, X, w):
+    r = y - X @ w
+    p_obj = (r @ r) / (2 * len(y))
+    return p_obj + alpha * np.sum(np.abs(w))
+
+
+@njit
+def primal_grp(alpha, y, X, w, grp_ptr, grp_indices):
+    r = y - X @ w
+    p_obj = (r @ r) / (2 * len(y))
+    for g in range(len(grp_ptr) - 1):
+        w_g = w[grp_indices[grp_ptr[g]:grp_ptr[g + 1]]]
+        p_obj += alpha * norm(w_g, ord=2)
+    return p_obj
+
+
+def gram_lasso(X, y, alpha, max_iter, tol, check_freq=10):
```

Comment:
@PABannier can you support warm start here and in gram_group_lasso() ?

Reply:
by that I mean add `w_init=None` as is done in celer_path, if `w = np.zeros(n_features) if w_init is None else w_init.copy()`